listener/avalon_listener/tcs_listener.py

```diff
@@ -104,6 +104,29 @@ def __init__(self, config):
         for m in rpc_methods:
             self.dispatcher.add_method(m)
 
+    def _valid_json_rpc_request(self, input_json):
+        """
+        Check whether input request is valid json
+        rpc request or not
+        Parameter
+            input_json is dict containing json rpc request
+        Returns
+            True if valid request
+            False otherwise
+        """
+        valid_keys = ["jsonrpc", "id", "method", "params"]
+        for k in input_json.keys():
+            if k not in valid_keys:
+                logger.info("Invalid parameter in json rpc request")
+                return False
+        if "jsonrpc" in input_json:
+            if input_json["jsonrpc"] != "2.0":
+                logger.info("Invalid json rpc version")
+                return False
+        if "id" in input_json:
+            return type(input_json["id"]) == int
```

Comment:
Please check https://www.jsonrpc.org/specification. An identifier established by the Client that MUST contain a String, Number, or NULL value if included. 

There are other error conditions for json rpc request which are not validated by this function. I would suggest to use library provided error handling than writing a custom error handling function for validating json rpc request. 
`response = JSONRPCResponseManager.handle()` returns a response object which will contain an error field in case of error. For invalid json rpc request,  `response.error` will look like this -> `{'code': -32600, 'message': 'Invalid Request'}`

Reply:
In our spec everywhere id is mentioned as integer. Then we should update our spec as well.